backend/backend/api/routes/room.py

```diff
@@ -0,0 +1,99 @@
+from functools import lru_cache
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, status, HTTPException
+
+from backend.common.room import Room
+from backend.common.player import Player
+from backend.api.models.room import RoomId, UpdateLeader
+from backend.api.models.player import PlayerId
+from backend.api.routes.player import get_players
+
+router = APIRouter(tags=["rooms"])
+
+
+@lru_cache
+def get_rooms() -> dict[str, Room]:
+    return {}
+
+
+@router.post("/rooms/", status_code=status.HTTP_201_CREATED,
+             response_model=RoomId)
+async def create_room(player_id: PlayerId,
+                      rooms: Annotated[dict[str, Room], Depends(get_rooms)],
+                      players: Annotated[dict[str, str], Depends(get_players)]):
+    room = Room()
+    player_name = players.get(player_id.player_id)
+
+    if not player_name:
+        raise HTTPException(detail=f"no user under given id {player_id.player_id}",
+                             status_code=status.HTTP_404_NOT_FOUND)
+
+    player = Player(id=player_id.player_id,
+                    name=player_name)
+    player.leader = True
+    room.add_player(player)
+    rooms.update({room.room_id: room})
+    return RoomId(room_id=room.room_id)
+
+
+@router.post("/enter_room/room_id",
+             status_code=status.HTTP_202_ACCEPTED)
+async def enter_room(room_id: RoomId,
+                     player_id: PlayerId,
+                     rooms: Annotated[dict[str, Room], Depends(get_rooms)],
+                     players: Annotated[dict[str, str], Depends(get_players)]):
+    room = rooms.get(room_id.room_id)
+    if not room:
+        return HTTPException(detail=f"no such room {room_id.room_id}",
+                             status_code=status.HTTP_404_NOT_FOUND)
+
+    player_in_room = room.players.get(player_id.player_id)
+    if player_in_room:
+        return
```

Comment:
Can be dependency, also, you should return some data if the status code is other 204
202 - ACCEPTED means we had started a certain process, I'm not sure you should use 202 rather a simply 200 OK since no background process is starting